[package] Modify Digraph to track predecessors

lobstergrindset opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61148
* #61147
* **#61146**

Track predecessors of nodes in DiGraph in order to enable cleaner dependency visualization code.

Differential Revision: [D29559682](https://our.internmc.facebook.com/intern/diff/D29559682)

Commit:
[package] Modify Digraph to track predecessors

[ghstack-poisoned]

Commit:
Update on "[package] Modify Digraph to track predecessors"


Track predecessors of nodes in DiGraph in order to enable cleaner dependency visualization code.

[ghstack-poisoned]

Lilyjjo commented:
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29559682).

Commit:
Update on "[package] Modify Digraph to track predecessors"


Track predecessors of nodes in DiGraph in order to enable cleaner dependency visualization code.

Differential Revision: [D29559682](https://our.internmc.facebook.com/intern/diff/D29559682)

[ghstack-poisoned]

Lilyjjo commented:
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29559682).
